content/process/git-flow.md

```diff
@@ -1,3 +1,95 @@
 # Git
 
-TODO
+This guide assumes that you are already familiar with the basic functionalities of `git`:
+
+* Cloning repositories
+* Checking branches out
+* Committing changes
+* Pushing
+* Pulling
+
+And it will walk you through slightly more advanced concepts that will come in handy when working on Optakt projects.
+
+## Rebasing
+
+The most important thing you'll likely need to master is _rebasing_.
+Whenever the base branch one of your pull requests is against gets updated with changes that conflict with yours, you will need to rebase your branch before it can be properly reviewed and merged.
+
+!!! warning "git merge"
+
+    We do not use `git merge` at Optakt, because it would result in a messier commit history.
+    Rebasing, and squashing commits before merging pull requests ensures that when looking at master, each commit corresponds to a single pull request.
+
+1. Fetch the latest version of the base branch you need to rebase against. In most cases this will be `master`.
+2. Run `git rebase <base branch>` from your local branch.
+3. This will make `git` rewind history and put your HEAD back to the first commit that diverges between the two branches.
+4. One by one, commits will be automatically applied to your local branch until one with conflicts is reached.
```

Comment:
It's also possible to `git rebase -i <commit-hash>` to squash commits in reference to a specific commit (e.g. all commits since commit X). It's handy if one updated the base/target branch first and then wants to squash commits to rebase/cherry pick..